modules/dnn/src/layers/resize2_layer.cpp

```diff
@@ -0,0 +1,1010 @@
+    std::vector<bool> outOfBoundsY(outH, false);
+    for (int oy = 0; oy < outH; ++oy)
+    {
+        float src_y = computeSrcGeneric(oy, scaleH, inH, lenY, coordMode, halfPixelCenters, start_y, end_y);
+        int iy = int(std::floor(src_y));
+        float dy = src_y - iy;
+        float swy = 0.f;
+        bool hasOutOfBounds = false;
+        interpolateCubicResize(dy, cubicA, y_w[oy].data());
+        if (!tf_crop_and_resize_mode && !excludeOutside)
+        {
+            for (int k = -1; k <= 2; ++k)
+            {
+                int idy = clampInt(iy + k, 0, inH - 1);
+                y_id[oy][k+1] = idy;
+                swy += y_w[oy][k+1];
+            }
+        }
+        else if (tf_crop_and_resize_mode)
+        {
+            for (int k = -1; k <= 2; ++k)
+            {
+                int idy = iy + k;
+                unsigned valid = (unsigned)idy < (unsigned)inH;
+                y_id[oy][k+1] = valid ? idy : -1;
+                float w = y_w[oy][k+1];
+                float wv = valid ? w : 0.f;
+                y_w[oy][k+1] = wv;
+                swy += wv;
+                hasOutOfBounds |= !valid;
+            }
+        }
+        else
+        {
+            for (int k = -1; k <= 2; ++k)
+            {
+                int idy = iy + k;
+                unsigned valid = (unsigned)idy < (unsigned)inH;
+                y_id[oy][k+1] = valid ? idy : -1;
+                float w = y_w[oy][k+1];
+                float wv = valid ? w : 0.f;
+                y_w[oy][k+1] = wv;
+                swy += wv;
+            }
+        }
+        if (swy != 0.f)
+            for (int k = 0; k < 4; ++k)
+                y_w[oy][k] /= swy;
+        if (tf_crop_and_resize_mode && hasOutOfBounds) {
+            outOfBoundsY[oy] = true;
+        }
+    }
+
+    const int nstripes = 16;
+    parallel_for_(Range(0, nstripes), [&](const Range& range) {
+        int row0 = range.start * (outH * numPlanes) / nstripes;
+        int row1 = range.end   * (outH * numPlanes) / nstripes - 1;
+        int plane0 = row0 / outH, plane1 = row1 / outH;
+        row0 %= outH;
+        row1 %= outH;
+
+        for (int p = plane0; p <= plane1; ++p)
+        {
+            int oy0 = (p == plane0) ? row0 : 0;
+            int oy1 = (p == plane1) ? row1 : outH - 1;
+            const float* inpBase = inpPlanes.ptr<float>(p * inH);
+            for (int oy = oy0; oy <= oy1; ++oy)
+            {
+                float* outRow = outPlanes.ptr<float>(p * outH) + oy * outW;
+
+                if (tf_crop_and_resize_mode && outOfBoundsY[oy]) {
+                    for (int ox = 0; ox < outW; ++ox)
+                        outRow[ox] = extrapolation_value;
+                    continue;
+                }
+
+                std::vector<float> hbuf(inW, 0.f);
+                for (int ix = 0; ix < inW; ++ix)
+                {
+                    float v = 0.f;
+                    bool hasValid = false;
+                    for (int ky = 0; ky < 4; ++ky)
```

Comment:
this loop should be replaced with `float v = ptr0[ix]*w0 + ptr1[ix]*w1 + ptr2[ix]*w2 + ptr3[ix]*w3;`